FileShareService.DesktopClient.Core/AuthProvider.cs

```diff
@@ -57,11 +58,25 @@ protected set
 
         [ExcludeFromCodeCoverage] // Can't unit test the login process as it is calling out to real AAD
         public async Task<string?> Login()
+        {
+            await GetToken();
+            IsLoggedIn = true;
+            return CurrentAccessToken;
+        }
+
+        public string? CurrentAccessToken => authenticationResult?.AccessToken;
+        public DateTimeOffset? CurrentAccessTokenExpiry => authenticationResult?.ExpiresOn;
+
+        public IEnumerable<string> Roles => IsLoggedIn && authenticationResult != null
+            ? jwtTokenParser.ParseRoles(authenticationResult.AccessToken)
+            : Enumerable.Empty<string>();
+
+        public async  Task<string> GetToken()
```

Comment:
To avoid hardcoding the login-url/authority, I think it is better to use:       `.WithAuthority(AzureCloudInstance.AzurePublic, tenantId)`

Similarly, to avoid hardcoding the redirect-url, I think it is better to use:  `.WithDefaultRedirectUri()`

So the `Build()` call becomes:
```
var publicClientApplication = PublicClientApplicationBuilder
                .Create(environmentsManager.CurrentEnvironment.ClientId)
                .WithAuthority(AzureCloudInstance.AzurePublic, tenantId)
                .WithDefaultRedirectUri()
                .Build();
```

Reply:
Resolved